### PR TITLE
feat[venom]: scheduler optimization

### PR DESCRIPTION
**Pull request by harkal**
### Commit message

Commit message for the final, squashed PR. (Optional, but reviewers will appreciate it! Please see [our commit message style guide](../../master/docs/style-guide.rst#best-practices-1) for what we would ideally like to see in a commit message.)

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: vyper/venom/passes/dft.py [hunk 1]
```diff
@@ -1,81 +1,255 @@
...
 from vyper.venom.function import IRFunction
 from vyper.venom.passes.base_pass import IRPass
 
+count = 0
+
+
+@dataclass
+class Group:
+    group_id: int
+    dependants: list["Group"]
```
Comment (charles-cooper) [inline vyper/venom/passes/dft.py:18]: sp: dependents

**Comment by charles-cooper** [comment]
@harkal is this superseded by #4255? if so i guess we can close this

**Comment by charles-cooper** [comment]
@harkal i think this is superseded by #4255 and we can close this?

**Comment by harkal** [comment]
> @harkal is this superseded by #4255? if so i guess we can close this

yes! I am closing it

**Pull request closed without merging**